### ISSUE TITLE
Panache reactive and Camel/SQS reactive-messaging consumer hangs on 4th message

### ISSUE THREAD

**Issue by dgian** [quarkusio/quarkus#25110]
### Describe the bug

Consuming messages from an SQS queue through Apache Camel + reactive-messaging and then persisting to PostgreSQL using Panache Reactive, hangs on the 4th message. Increasing the pool size, increases the amount of messages consumed before hanging.

Consumer  code:

```
@Incoming("persons")
public Uni<Void> consume(String payload) {
    return Uni.createFrom()
        .item(payload)
        .map(json -> Json.decodeValue(json, Person.class))
        .flatMap(repository::save)
        .onFailure().invoke(t -> log.error(t.getMessage(), t));
}
```

Repository code:

```
@ReactiveTransactional
public Uni<Void> save(Person person) {
    return persist(person)
        .invoke(() -> log.info("Saved person {}", person.getFirstName()))
        .replaceWithVoid();
}
```

**Note that** this is reproducible with Quarkus 2.7.1 and later (up to 2.8.2). Downgrading to Quarkus 2.7.0 makes this issue not reproducible.

_**Possibly related to**_  #23804 

### Expected behavior

Should be able to handle more than 4 messages without increasing the datasource pool size.

### Actual behavior

On 4th consumed message the stream **times out** with the following error:

```
io.smallrye.reactive.messaging.providers.SubscriberMediator reportFailure
ERROR: SRMSG00201: Error caught while processing a message 
io.vertx.core.impl.NoStackTraceThrowable: Timeout

io.smallrye.reactive.messaging.providers.SubscriberMediator$1 onError
ERROR: SRMSG00201: Error caught while processing a message
io.vertx.core.impl.NoStackTraceThrowable: Timeout
```

no other stacktrace is thrown apart from Apache Camel subscription cancel error.

Using a `sessionFactory.withTransaction` instead of panache reactive will throw a different kind of error:

```
Caused by: java.lang.IllegalStateException: The current operation requires a safe (isolated) Vert.x sub-context, but the current context hasn't been flagged as such. You can still use Hibernate Reactive, you just need to avoid using the methods which implicitly require accessing the stateful context, such as MutinySessionFactory#withTransaction and #withSession.
	at io.quarkus.vertx.core.runtime.context.VertxContextSafetyToggle.checkIsSafe(VertxContextSafetyToggle.java:80)
	at io.quarkus.vertx.core.runtime.context.VertxContextSafetyToggle.validateContextIfExists(VertxContextSafetyToggle.java:63)
	at io.quarkus.hibernate.reactive.runtime.customized.CheckingVertxContext.put(CheckingVertxContext.java:40)
	at org.hibernate.reactive.mutiny.impl.MutinySessionFactoryImpl.lambda$withSession$17(MutinySessionFactoryImpl.java:251)
	at io.smallrye.mutiny.groups.UniOnItem.lambda$invoke$0(UniOnItem.java:59)
	at io.smallrye.context.impl.wrappers.SlowContextualConsumer.accept(SlowContextualConsumer.java:21)
	at io.smallrye.mutiny.operators.uni.UniOnItemConsume$UniOnItemComsumeProcessor.invokeEventHandler(UniOnItemConsume.java:77)
	... 121 more
```

### How to Reproduce?

1. Clone repo https://github.com/dgian/qpr-reproduce-sample
2. Run the automated test: `./gradlew :test --tests "sample.ConsumerTest"` (or through IDE)

### Output of `uname -a` or `ver`

5.15.32-1-MANJARO #1 SMP PREEMPT Mon Mar 28 09:16:36 UTC 2022 x86_64 GNU/Linux

### Output of `java -version`

openjdk 17.0.3 2022-04-19 OpenJDK Runtime Environment (build 17.0.3+3) OpenJDK 64-Bit Server VM (build 17.0.3+3, mixed mode)

### Quarkus version or git rev

2.8.2 (but reproduced from 2.7.1)

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 7.4.2

**Comment by FroMage** [comment]
This sounds like a request scoped session hanging on forever, no? Or something else @cescoffier @DavideD ?

**Comment by cescoffier** [comment]
The camel connector does not create duplicated contexts yet, so it cannot use hibernate reactive safely.

**Comment by dgian** [comment]
Thank you @cescoffier !
Should this be expected in the next few releases? Or is it low priority?

**Comment by cescoffier** [comment]
I cannot really tell. My plate is full at the moment, so it will have to wait until I'm done with one of my tasks. Hopefully end of June should be fine.

Of course, if someone beats me, it's all fine.

**Comment by uPagge** [comment]
Will there be a solution to this problem?
Perhaps there are some workarounds?

**Comment by uPagge** [comment]
Is there any plan to fix the original issue so that Panache can be used easily? Or is it basically impossible?

**Comment by DavideD** [comment]
For reference, the time out issue with Panache is because the [ReactiveSessionProducer dispose method](https://github.com/quarkusio/quarkus/blob/505770c3c3b89507f2f97ef9a5e0b32f029e3239/extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/ReactiveSessionProducer.java#L27) doesn't get called at the right time and connection never released.

I haven't run the test case but we have seen this same error for https://github.com/quarkusio/quarkus/issues/23739 and [when using a scheduler](https://github.com/quarkusio/quarkus/pull/27108)

**Comment by FroMage** [comment]
OK, so this is a duplicate of #23739 no?

**Comment by DavideD** [comment]
I don't think so. This issue mentions two errors:
1. Timeout after 4 requests: this might or might not be solved by #23739. I haven't checked, but it depends whose managing the context in this scenario
2. Unsafe vert.x sub-context: I don't think #23739 affect this.
